### PR TITLE
Use null coalescing operator

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

There were a few comments in PRs to use null coalescing operator ... here we go.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Review by Sdfendor — APPROVED** [review]
LGTM, great effort!

**Comment by sreichel** [comment]
> did you do this manually?

PHPstorm shows possible usage of null coalescing operator ... i've run inspection for only that and changed that half-automated. (Y, i've checked every change)

**Review by fballiano — APPROVED** [review]
i've no way of testing all this so i'll just trust the process :-)

File: app/code/core/Mage/Rule/Model/Condition/Combine.php [hunk 1]
```diff
@@ -255,10 +255,8 @@ public function asXml($containerKey = 'conditions', $itemKey = 'condition')
      */
     public function loadArray($arr, $key = 'conditions')
     {
-        $this->setAggregator(isset($arr['aggregator']) ? $arr['aggregator']
-                : (isset($arr['attribute']) ? $arr['attribute'] : null))
-            ->setValue(isset($arr['value']) ? $arr['value']
-                : (isset($arr['operator']) ? $arr['operator'] : null));
+        $this->setAggregator($arr['aggregator'] ?? ($arr['attribute'] ?? null))
+            ->setValue($arr['value'] ?? ($arr['operator'] ?? null));
```
Comment (kiatng) [inline app/code/core/Mage/Rule/Model/Condition/Combine.php:259]: Is this better?
```suggestion
        $this->setAggregator($arr['aggregator'] ?? $arr['attribute'] ?? null)
            ->setValue($arr['value'] ?? $arr['operator'] ?? null);
```

File: app/code/core/Mage/Rule/Model/Condition/Combine.php [hunk 2]
```diff
@@ -255,8 +255,8 @@ public function asXml($containerKey = 'conditions', $itemKey = 'condition')
      */
     public function loadArray($arr, $key = 'conditions')
     {
-        $this->setAggregator($arr['aggregator'] ?? ($arr['attribute'] ?? null))
-            ->setValue($arr['value'] ?? ($arr['operator'] ?? null));
+        $this->setAggregator($arr['aggregator'] ?? $arr['attribute'] ?? null)
+            ->setValue($arr['value'] ?? $arr['operator'] ?? null);
```
Comment (fballiano) [inline app/code/core/Mage/Rule/Model/Condition/Combine.php:259]: IMHO this is hard to understard, extremely short code is already less easy on the brain, but this (to me) it's a no. with the parenthesis was a bit better

Comment (sreichel) [reply app/code/core/Mage/Rule/Model/Condition/Combine.php:259]: imho ... for me its easier to read w/o parenthesis.  I can revert it later if you think it was better  before.

Comment (kiatng) [reply app/code/core/Mage/Rule/Model/Condition/Combine.php:259]: Ref https://www.php.net/manual/en/migration70.new-features.php, the operator can be chained. With parenthesis, it defeats the feature.

**Review by kiatng — APPROVED** [review]

File: app/code/core/Mage/Rule/Model/Condition/Combine.php (hunk 2 above)
Comment (Sdfendor) [reply app/code/core/Mage/Rule/Model/Condition/Combine.php:259]: Chaining seems to me a good argument against the additional parentheses.

**Review by Sdfendor — APPROVED** [review]

**Review by kiatng — APPROVED** [review]

**Review by Sdfendor — APPROVED** [review]

**Review by ADDISON74 — APPROVED** [review]

**Pull request merged**